### ISSUE TITLE
Store: Clicking on "Stats" hides "Store" from interface

### ISSUE THREAD

**Issue by iamgabrielma** [Automattic/wp-calypso#21554]
Once a Store is created, visiting your website logged in (not from WordPress.com but from the website itself) clicking on "Stats" tab within calypso hides the "Store" tab from the interface, making it harder to go back to manage my products or find where I'm supposed to go to set everything up.

#### Steps to reproduce
1. Starting at any Store, click on `My Sites`on the WP toolbar to see the Calypso menu.
<img width="1279" alt="screen shot 2018-01-16 at 17 04 20" src="https://user-images.githubusercontent.com/3812076/34983195-53fe15de-fadf-11e7-956f-1ffa2fa51182.png">
2. Click "Stats"
3. You're redirected to https://wordpress.com/store/stats/orders/day/`$DOMAIN`
4. This hides the Store button and its options cannot be accessed directly anymore:
<img width="1277" alt="screen shot 2018-01-16 at 09 03 46" src="https://user-images.githubusercontent.com/3812076/34983289-98fa236c-fadf-11e7-836c-70638e184ae2.png">

#### What I expected

- To be able to see `Store` button all the time and manage my products and store from Calypso

#### What happened instead

- The button disappears, clicking on `gridicons-arrow-left` (what the user would do to go back to the previous menu) or Switch site just shows me the sites under my account, but clicking on them does not bring me to the starting point where I can see the Store button.
- Clicking on any other tab does not bring back the Store button unless I type manually the URL  `https://wordpress.com/store/$Store` to go back to the Store Management page.

**Comment by timmyc** [comment]
Thanks for the detailed issue @iamgabrielma! I _think_ what might be happening right now could possibly be some issues with GEO IP potentially as I couldn't recreate the issue myself.

The Store sidebar link is set to get an overhaul in #21466, and likely a follow-up PR for that - so I'm going to leave this open for now but would love to have you test out this flow again once that branch is merged.

**Comment by timmyc** [comment]
I'm still not able to re-create this issue myself, and since there have been no other reports, I'm going to close this out.
